dhis-2/dhis-api/src/main/java/org/hisp/dhis/appmanager/AppManager.java

```diff
@@ -240,4 +209,63 @@ Resource getAppResource( App app, String pageName )
      */
     boolean markAppToDelete( App app );
 
+    // -------------------------------------------------------------------------
+    // Static methods for manipulating a collection of apps
+    // -------------------------------------------------------------------------
+
+    /**
+     * Returns a list of all installed apps with AppType equal the given Type
+     *
+     * @return list of installed apps with given AppType
+     */
+    public static List<App> filterAppsByType( AppType appType, Collection<App> apps )
+    {
+        return apps.stream()
+            .filter( app -> app.getAppType() == appType )
+            .collect( Collectors.toList() );
+    }
+
+    /**
+     * Returns a list of all installed apps with name equal the given name and
+     * operator. Currently supports eq and ilike.
+     *
+     * @return list of installed apps with given name
+     */
+    public static List<App> filterAppsByName( final String name, Collection<App> apps, final String operator )
+    {
+        return apps.stream().filter(
+            app -> (("ilike".equalsIgnoreCase( operator ) && app.getName().toLowerCase().contains( name.toLowerCase() ))
+                ||
+                ("eq".equalsIgnoreCase( operator ) && app.getName().equals( name ))) )
+            .collect( Collectors.toList() );
+    }
+
+    /**
+     * Returns a list of all installed apps with shortName equal the given name
+     * and operator. Currently supports eq and ilike.
+     *
+     * @return list of installed apps with given name
+     */
+    public static List<App> filterAppsByShortName( final String name, Collection<App> apps, final String operator )
+    {
+        return apps.stream()
+            .filter( app -> (("ilike".equalsIgnoreCase( operator )
+                && app.getShortName().toLowerCase().contains( name.toLowerCase() )) ||
+                ("eq".equalsIgnoreCase( operator ) && app.getShortName().equals( name ))) )
+            .collect( Collectors.toList() );
```

Comment:
Very similar to the above method "filterAppsByName()", maybe break it down to a common method that use name, app property and operator as input.